src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncMethodBuilderMemberCollection.cs

```diff
@@ -345,10 +345,20 @@ private static NamedTypeSymbol ValidateBuilderType(SyntheticBoundNodeFactory F,
                  !builderType.IsVoidType() &&
                  (forMethodLevelBuilder || builderType.DeclaredAccessibility == desiredAccessibility))
             {
-                bool isArityOk = isGeneric
-                                 ? builderType.IsUnboundGenericType && builderType.ContainingType?.IsGenericType != true && builderType.Arity == 1
-                                 : !builderType.IsGenericType;
-                if (isArityOk)
+                if (isGeneric)
```

Comment:
Could you please clarify the difference between `isGeneric` and `builderType.IsGenericType`?

Reply:
Is it something like: `isGeneric` refers to whether the task-like thing is generic, while `builderType.IsGenericType` refers to whether the builder is generic?

Reply:
Correct. Whether the return type is generic or not drives the rest of the process.